packages/vuetify/src/effects/color.ts

```diff
@@ -0,0 +1,83 @@
+import { computed, ComputedRef, isRef } from 'vue'
+
+export interface ColorProps {
+  color?: string
+}
+
+export const colorProps = (defaults: Partial<ColorProps> = {}) => ({
+  color: {
+    type: String,
+    default: defaults.color,
+  },
+})
+
+const isCssColor = (color?: string | false): boolean => !!color && !!color.match(/^(#|(rgb|hsl)a?\()/)
+const isProps = <T>(props: T | ComputedRef<string>): props is T => !isRef(props)
+
+type Color = { class: Record<string, boolean> } | { style: Record<string, string> } | {}
+
+type TextColor = {
+  textColor: ComputedRef<Color>
+}
+
+export function useTextColor (props: ColorProps | ComputedRef<string>): TextColor
```

Comment:
`Readonly<Ref>` for usage with normal (not computed) refs? And `ColorPropValue` for a bit more DRY

```ts
type ColorPropValue = string | undefined
export function useTextColor (props: ColorProps | Readonly<Ref<ColorPropValue>>): TextColor

useTextColor(computed(() => 'black'))
useTextColor(ref(undefined))
```